core/templates/dev/head/pages/exploration_editor/EditorServicesSpec.js

```diff
@@ -17,6 +17,106 @@
  *   editor page.
  */
 
+
+describe('Exploration data service', function() {
+  beforeEach(module('oppia'));
+
+  describe('getData local save', function() {
+    var eds = null;
+    var mockBackendApiService = null;
+    var mockLocalStorageService = null;
+    var mockUrlService = null;
+    var draftChangesValid = null;
+    var draftChangesInvalid = null;
+    var draftChangesInvalid = null;
+    var $q = null;
+
+    beforeEach(function() {
+      module(function($provide) {
+        $provide.value(
+          'LocalStorageService', mockLocalStorageService);
+      });
+      module(function($provide) {
+        $provide.value(
+          'EditableExplorationBackendApiService', mockBackendApiService);
+      });
+      module(function($provide) {
+        $provide.value(
+          'urlService', mockUrlService);
+      });
+    });
+
+    beforeEach(function() {
+      mockUrlService = {
+        getPathname: function() {}
+      };
+
+      mockBackendApiService = {
+        fetchApplyDraftExploration: function() {}
+      };
+
+      mockLocalStorageService = {
+        getExplorationDraft: function() {},
+        removeExplorationDraft: function() {}
+      };
+      spyOn(mockUrlService, 'getPathname').and.returnValue('/create/exp_id');
+    });
+
+    beforeEach(inject(function($injector) {
+      eds = $injector.get('explorationData');
+      $q = $injector.get('$q');
+    }));
+
+    beforeEach(function() {
+      expDataResponse = {
+        draft_change_list_id: 3,
+      };
+
+      draftChangesValid = {
```

Comment:
responseWhenDraftChangesAreValid

(There should be a noun describing what the thing "is" in the name of the object.)

Reply:
Done.